### PR TITLE
Catch and display load errors for WorkloadsOverview

### LINKED ISSUE #4344: One inaccessible namespace blocks Workloads Overview for all namespaces
https://user-images.githubusercontent.com/40840436/141815807-f7d25fd1-2610-4691-89a1-9fec6330e371.mov

In this example the user can list namespaces but only has access to default namespace. If default and any other namespaces are selected in the namespace filter then the statistics all show 0

similar to #4192

### PR DESCRIPTION
**Pull request by Nokel81**
- Move the select filter to the top level since that is where the
  subscribe call is

Signed-off-by: Sebastian Malton <sebastian@malton.name>

fixes #4344

### CODE REVIEW

**Review by jim-docker — COMMENTED** [review]
Something fishy:
Screen Shot 2021-11-19 at 2 42 21 PM

User does not have namespace list permission, accessible namespaces are set to `default`, `freddy`, and `garbage`. User has access only to `default`, `garbage` namespace does not exist. 

With just `default` selected I would not expect any warnings

**Comment by Nokel81** [comment]
@jim-docker Good catch, thanks. I also fixed the same issue on the `<KubeObjectListLayout />` because it exists there too.

File: src/extensions/registries/workloads-overview-detail-registry.ts [hunk 1]
```diff
@@ -19,22 +19,29 @@
...
   getItems() {
-    const items = super.getItems();
+    return orderBy(super.getItems(), "priority", "desc");
+  }
+
+  protected getRegisteredItem(item: WorkloadsOverviewDetailRegistration): RegisteredWorkloadsOverviewDetail {
+    const { priority = 50, ...rest } = item;
 
-    return items.sort((a, b) => (b.priority ?? 50) - (a.priority ?? 50));
+    return { priority, ...rest };
```
Comment (jim-docker) [inline src/extensions/registries/workloads-overview-detail-registry.ts:45]: this looks unrelated to the issue, did you test that it still works?

File: src/renderer/components/+workloads-overview/overview.tsx [hunk 2]
```diff
@@ -33,36 +33,77 @@ import { replicaSetStore } from "../+workloads-replicasets/replicasets.store";
...
     disposeOnUnmount(this, [
       kubeWatchApi.subscribeStores([
         podsStore, deploymentStore, daemonSetStore, statefulSetStore, replicaSetStore,
         jobStore, cronJobStore, eventStore,
       ], {
-        preload: true,
-        namespaces: clusterContext.contextNamespaces,
+        onLoadFailure: error => this.loadErrors.push(String(error)),
+      }),
+      reaction(() => clusterContext.contextNamespaces.slice(), () => {
```
Comment (jim-docker) [inline src/renderer/components/+workloads-overview/overview.tsx:64]: Why can't it just be `() => clusterContext.contextNamespaces`?

**Review by jim-docker — APPROVED** [review]

File: src/extensions/registries/workloads-overview-detail-registry.ts (hunk 1 above)
Comment (Nokel81) [reply src/extensions/registries/workloads-overview-detail-registry.ts:45]: Yes I did. With the kube-resource-map extension installed (and the default overview doesn't have `priority` defined) the order is the exact same as before.

File: src/renderer/components/+workloads-overview/overview.tsx (hunk 2 above)
Comment (Nokel81) [reply src/renderer/components/+workloads-overview/overview.tsx:64]: Because then it only tracks assignments to the `clusterContext.contextNamespaces` field itself, and not internal changes.

Comment (jim-docker) [reply src/renderer/components/+workloads-overview/overview.tsx:64]: Ah, right. You have to "dot" into it. Ugh.
